common/djangoapps/student/helpers.py

```diff
@@ -368,6 +368,24 @@ def get_redirect_to(request):
     return redirect_to
 
 
+def _is_safe_redirect(request, redirect_to):
+    """
+    Determine if the given redirect URL/path is safe for redirection.
+
+    The redirect_to URL must be a safe URL. The redirect URL must also
+    be either the same domain or a subdomain of the request host domain.
+    """
+    request_host = request.get_host()
+    request_host_domain = '.'.join(request_host.split('.')[-2:])
+    redirect_to_host = urlparse.urlsplit(redirect_to).netloc
+    return (
+        # Use wildcard check to ensure redirect to the same domain or a subdomain
+        http.is_same_domain(urlparse.urlsplit(redirect_to).netloc, '.' + request_host_domain) and
+        # Checks various parts and formats of the redirect URL to ensure it is safe.
+        http.is_safe_url(redirect_to, allowed_hosts={redirect_to_host}, require_https=True)
+    )
```

Comment:
Can we make the innards of this method more readable?  For example, use intermediate variables to explain the "magic" around the parsing?  Thanks.

Reply:
@nasthagiri: Refactored.  Let me know if you approve.